### PR TITLE
Adding User Data Endpoints

### LINKED ISSUE #12: [Endpoints] User data endpoints
Create the user data endpoints. All the endpoints should work under ''/users" route. The most important endpoints should be:

1. Check user token validity
2. Get user data
3. Get user's appointments
4. Get user's analyses
5. Update user data

Future endpoints will be added

### PR DESCRIPTION
**Pull request by omerdikyol**
1. Adding User Data Endpoints to User Controller and User Route.
2. Routing Endpoints to /users/
3. Creating Register Login Endpoints to test User Data Endpoints. (testing purposes)
4. Adding Hash Function for Register and Login Endpoints (testing purposes) 5.Removing unnecessary .placeholder files.
6. Adding Specilization variable to User Model.
7. Adding Results Model taken from M3 ( to test Get Results Endpoint ).
8. Creating Simple Response interface. 
Note: Endpoints will not work without authorization and JWT Validation. So it will work after that task is closed.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -6,11 +6,15 @@ const app = express();
 app.use(express.json())
 app.use(express.urlencoded({ extended: true }))
 
+// routes
+import userRouter from "./routes/user";
```
Comment (LucaStefan112) [inline src/index.ts:7]: imports at the top of the file

File: src/index.ts [hunk 2]
```diff
@@ -6,11 +6,15 @@ const app = express();
...
 
 app.get("/", (req, res) => {
```
Comment (LucaStefan112) [inline src/index.ts]: delete this endpoint

File: src/models/results.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import mongoose from "mongoose";
+export interface IResults{
```
Comment (LucaStefan112) [inline src/models/results.ts]: change Result to Analysis -> also change to singular

File: src/controllers/user.controller.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        return res.status(404).send({ success: false, message: "User not found" });
+    }
+
+    const data = {
+        fullName: user.fullName,
+        email: user.email,
+        role: user.role,
+        dateOfBirth: user.dateOfBirth,
+        phoneNumber: user.phoneNumber,
+        specialization: user.specialization,
```
Comment (MariusAchitei) [inline src/controllers/user.controller.ts:31]: Sugi pula

File: src/controllers/user.controller.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ import jwt from "jsonwebtoken";
 // models
 import User from "../models/user";
 import Appointment from "../models/appointment";
-import Result from "../models/results";
+import Analyse from "../models/analysis";
```
Comment (LucaStefan112) [inline src/controllers/user.controller.ts]: Change spelling to Analysis

File: src/controllers/user.controller.ts [hunk 6]
```diff
@@ -53,10 +52,10 @@ export const getAppointments = async (req: Request, res: IBasicResponse) => {
     }
 }
 
-export const getResults = async (req: Request, res: IBasicResponse) => {
+export const getAnalyses = async (req: Request, res: IBasicResponse) => {
     const userId = req.body.user._id;
     try {
-        const results = await Result.find({ person: userId });
+        const results = await Analyse.find({ person: userId });
```
Comment (LucaStefan112) [inline src/controllers/user.controller.ts]: change spelling to Analysis

**Review by LucaStefan112 — CHANGES_REQUESTED** [review]

**Review by LucaStefan112 — APPROVED** [review]

**Pull request merged**